src/transformers/models/ernie_m/modeling_ernie_m.py

```diff
@@ -77,15 +77,14 @@ def forward(
             inputs_embeds = self.word_embeddings(input_ids)
         if position_ids is None:
             input_shape = inputs_embeds.size()[:-1]
-            ones = torch.ones(input_shape, dtype=torch.int64)
+            ones = torch.ones(input_shape, dtype=torch.int64, device=inputs_embeds.device)
             seq_length = torch.cumsum(ones, dim=1)
             position_ids = seq_length - ones
 
             if past_key_values_length > 0:
                 position_ids = position_ids + past_key_values_length
         # to mimic paddlenlp implementation
         position_ids += 2
-        position_ids = position_ids.to(next(self.position_embeddings.parameters()).device)
```

Comment:
With `nn.DataParallel`, we get `E       StopIteration: Caught StopIteration in replica 1 on device 1.` See [this](https://github.com/huggingface/transformers/blob/03aaac35021dead4fb7ad354fe9c986d16869f03/src/transformers/modeling_utils.py#L150).



Let's avoid using this `next(self.position_embeddings.parameters()).device` if possible.

Reply:
Yes, this is bad practice in general and will also not work with Accelerate big model inference.